### ISSUE TITLE
How to  adapt event triggered camera upload pre-video-stream behavier?

### ISSUE THREAD

**Issue by awsa2ron** [awslabs/amazon-kinesis-video-streams-producer-c#54]
Hi,
Background is that the camera triggered by some outside event, like FIR detecting human or pet. But somehow there will be some delay between human enter to create and upload kvs data, at most 10s in some cases.
So we hold another video buffer for those frames which already past, and upload those past video frame with absolute TS.

The issue is that, event trigger is random. when event continue with last event without gap, some video data's TS overlay previous and broken playback. log is in the attachment.

I think this is an important feature of kvs sdk. Could you help to review the log and we will appreciate if you can give more suggestion and solutions.

Thanks.

**Comment by MushMal** [comment]
@awsa2ron I don't see the attached logs. How does it break the playback? I assume you are using GetMedia or HLS/MPEG-DASH? 

TL;DR

There are multiple solutions that one could think.

* Have your pre-buffer to manage the timestamps so it can de-dup the frames in case of the overlapped sessions.
* Overlaps are OK if you are using GetMedia and querying with the Producer timestamps.
* Don't use intermittent producer. Have a single stream crated at the boot of the device. As you get frames submit them directly into the KVS stream. The latest codebase (in PIC) should be able to handle intermittent producer scenario automatically.

**Comment by awsa2ron** [comment]
Hi, 
After struggling few days on this topic, we found that one of solution is keep video data timestamp unique, even if the content is same to avoid concurrency on kvs server.

I will close this.
